src/services/SendEmailProcessor.ts

```diff
@@ -1,46 +1,61 @@
+import { ServicesEnum } from "../models/enums/ServicesEnum";
 import { ValidationHelper } from "../utils/ValidationHelper";
+import { createDynamoDbClient } from "../utils/DynamoDBFactory";
+import { buildCoreEventFields } from "../utils/TxmaEvent";
 import { Logger } from "@aws-lambda-powertools/logger";
 import { Metrics } from "@aws-lambda-powertools/metrics";
 import { SendEmailService } from "./SendEmailService";
+import { IPRService } from "./IPRService";
+import { EnvironmentVariables } from "./EnvironmentVariables";
 
 export class SendEmailProcessor {
 
-    private static instance: SendEmailProcessor;
+	private static instance: SendEmailProcessor;
 
-    private readonly logger: Logger;
+	private readonly logger: Logger;
 
-    private readonly metrics: Metrics;
+	private readonly metrics: Metrics;
 
-    private readonly validationHelper: ValidationHelper;
+	private readonly environmentVariables: EnvironmentVariables;
+
+	private readonly validationHelper: ValidationHelper;
 
 	private readonly govNotifyService: SendEmailService;
 
-	constructor(logger: Logger, metrics: Metrics, GOVUKNOTIFY_API_KEY: string) {
+	private readonly iprService: IPRService;
 
-    	this.logger = logger;
-    	this.validationHelper = new ValidationHelper();
-    	this.metrics = metrics;
+	constructor(logger: Logger, metrics: Metrics, GOVUKNOTIFY_API_KEY: string) {
+		this.logger = logger;
+		this.validationHelper = new ValidationHelper();
+		this.metrics = metrics;
 		this.govNotifyService = SendEmailService.getInstance(this.logger, GOVUKNOTIFY_API_KEY);
+		this.environmentVariables = new EnvironmentVariables(logger, ServicesEnum.SEND_EMAIL_PROCESSOR_SERVICE);
+		this.iprService = IPRService.getInstance(this.environmentVariables.sessionEventsTable(), this.logger, createDynamoDbClient());
 	}
 
 	static getInstance(logger: Logger, metrics: Metrics, GOVUKNOTIFY_API_KEY: string): SendEmailProcessor {
-    	if (!SendEmailProcessor.instance) {
-    		SendEmailProcessor.instance = new SendEmailProcessor(logger, metrics, GOVUKNOTIFY_API_KEY);
-    	}
-    	return SendEmailProcessor.instance;
+		if (!SendEmailProcessor.instance) {
+			SendEmailProcessor.instance = new SendEmailProcessor(logger, metrics, GOVUKNOTIFY_API_KEY);
+		}
+		return SendEmailProcessor.instance;
 	}
 
 	async processRequest(eventBody: any): Promise<EmailResponse> {
 		const email = Email.parseRequest(JSON.stringify(eventBody.Message));
 
-    	await this.validationHelper.validateModel(email, this.logger);
+		await this.validationHelper.validateModel(email, this.logger);
+
+		const emailResponse: EmailResponse = await this.govNotifyService.sendEmail(email);
 
-    	const emailResponse: EmailResponse = await this.govNotifyService.sendEmail(email);
-    	this.logger.debug("Response after sending Email message", { emailResponse });
+		await this.iprService.sendToTXMA({
+			event_name: "IPR_RESULT_NOTIFICATION_EMAILED",
+			...buildCoreEventFields({}),
```

Comment:
could you check with Sanka please if adding email address as per [this RFC](https://github.com/alphagov/digital-identity-architecture/blob/main/rfc/0023-crossteam-audit-and-analysis-eventmessage-payload.md) for IPR_RESULT_NOTIFICATION_EMAILED event would be useful

Reply:
Sure, asked him

Reply:
He has confirmed we want email so I've added that. He has also confirmed that we need user_id so having a discussion offline about how we might get access to that here

Reply:
Agreed that user_id will come later as there is more work required to get it